tests/run-make/print-target-list/rmake.rs

```diff
@@ -1,10 +1,15 @@
-// Checks that all the targets returned by `rustc --print target-list` are valid
-// target specifications
+// Checks that all the targets returned by `rustc --print target-list` are valid target
+// specifications.
+
+// ignore-tidy-linelength
+//@ needs-llvm-components: aarch64 arm avr bpf csky hexagon loongarch m68k mips msp430 nvptx powerpc riscv sparc systemz webassembly x86
+// FIXME(jieyouxu): there has to be a better way to do this, without the needs-llvm-components it
+// will fail on LLVM built without all of the components listed above.
```

Comment:
I'm not a fan of this, but I also am not sure if there's a better way

Reply:
Maybe `rustc` should be filtering this before printing targets anyway? Or else it could do a better job of avoiding LLVM initialization when it's not really needed -- I see no reason why `--print sysroot` should care.

That might also help https://github.com/coreos/cargo-vendor-filterer/issues/87